Appium 1.5

alizelzele opened the pull request:
update to support appium 1.5, replacing javascripts in `SelionAppiumIosDriver` with tap and touch actions.

Fixes #300

mach6 commented on pom.xml:
```diff
@@ -286,6 +286,11 @@
                         </execution>
                     </executions>
                 </plugin>
+                <plugin>
```
Why is this plugin needed? Please remove it.

mach6 commented on client/src/test/java/com/paypal/selion/android/sample/AppiumAndroidCheckBoxSeekBarTest.java:
```diff
@@ -40,22 +40,22 @@
 
-    @Test
-    @MobileTest(appPath = PAGE_OBJECTS_APP_PATH, device = DEVICE_NAME)
+//    @Test
```
Why is this commented out?

alizelzele replied:
I comment tests to run only one test and see the result, it might be commented out by mistake, the test is passed successfully though. Should it be commented again ?

mach6 replied:
yes. please remove the comment.

sebady replied:
I will fix this during the merge and address the conflict with the parent POM.

mach6 commented on client/src/main/java/com/paypal/selion/platform/grid/SeLionAppiumIOSDriver.java:
```diff
@@ -76,64 +64,71 @@ public WebElement findElementBy(By by) {
+        String s = gestureOptions.get(GestureOptions.TOUCH_COUNT);
+        int touchCount = s == null ? 1 : Integer.parseInt(s);
+        s = gestureOptions.get(GestureOptions.DURATION);
+        int duration = s == null ? 100 : Integer.parseInt(s);
```
Please code the default values to `static final` constant values.

mach6 commented on client/src/main/java/com/paypal/selion/platform/grid/SeLionAppiumIOSDriver.java:
```diff
@@ -76,64 +64,71 @@ public WebElement findElementBy(By by) {
-        javaScriptExecutor.executeScript(TAP_SCRIPT, optionsMap);
+        new TouchAction(this)
+                .tap(webElement).release()
+                .waitAction(100)
```
What makes `100` and acceptable value here? It seems awfully like `magic number` in this context. Can we fix this?

mach6 commented on client/src/main/java/com/paypal/selion/internal/platform/grid/browsercapabilities/AppiumCapabilitiesBuilder.java:
```diff
@@ -30,8 +30,10 @@
+    private static final String MOBILE_APP_MAIN_ACTIVITY = "appActivity";
+    private static final String MOBILE_APP_WAIT_ACTIVITY = "appWaitActivity";
+    private static final String MOBILE_APP_PACKAGE = "appPackage";
+    private static final String MOBILE_APP_WAIT_PACKAGE = "appWaitPackage";
```
Can you please point to some documentation on these 4 capabilities? I'm trying to understand their significance. Thanks!

alizelzele replied:
@mach6 appium 1.5.x with some of the apks, appium can not read the manifest and throws exception [1.5.2 - Android fails](https://github.com/appium/appium/issues/6499). the way to solve this issue is by defining package and main activity of the application. i was not able to find the official document to reference to, but if you run appium ui and set these 2 parameters, it will print in console:

```
--app-pkg => --default-capabilities '{"appPackage":"com.paypal.selion.pageobjectsdemoapp"}'
--app-activity => --default-capabilities '{"appActivity":".PageObjectsHomeActivity"}'
```

2 other parameters are, wait for the same package and activity to start :)

mach6 commented on client/src/main/java/com/paypal/selion/platform/grid/SeLionAppiumIOSDriver.java:
```diff
@@ -76,64 +64,71 @@ public WebElement findElementBy(By by) {
-        arguments.put(ELEMENT, elementId);
-        javaScriptExecutor.executeScript(SCROLLTO_SCRIPT, arguments);
+        Dimension dimension = manage().window().getSize();
+        int height = (int) (dimension.getHeight() * 0.7);
```
`0.7` seems to be a `magic number` ... What is the significance?  How does this work?  A code comment could help. :)

alizelzele replied:
sorry for all those magic numbers. I tried to clean them a bit and use meaningful const instead.
in this case, we need some space from around the edge, to not activate another functionalities of IOS instead of scrolling. :)

Commit:
support appium 1.5

1. update java_client
2. remove javascripts from appium ios driver
3. add appPackage and appMainActivity which are required for android in appium 1.5
4. cleaning and optimizing suites a bit

sebady commented on client/src/main/java/com/paypal/selion/configuration/Config.java:
```diff
@@ -505,6 +505,16 @@ public static synchronized void setConfigProperty(ConfigProperty configProperty,
         /**
+         * This parameter represents the main activity of Android mobile appplication. By default it is always empty.
+         */
+        MOBILE_APP_MAIN_ACTIVITY("mobileAppMainActivity","", false),
```
Since these new properties are Android only can we add "Android" to the property names here?

alizelzele replied:
@sebady Good idea 👍 
Updated the code according to.

mach6 reviewed (commented):
@alizelzele Please rebase and address any open comments. For the most part this PR LGTM. I'd like to get it merged soon. Also fyi, @sebady is reviewing this too.

sebady commented:
merged
